device/transport/amqp/test/_amqp_test.js

```diff
@@ -259,6 +259,142 @@ describe('Amqp', function () {
+          assert(fakeBaseClient.attachReceiverLink.calledWith(transport._deviceMethodClient._methodEndpoint));
+          testCallback();
+        });
+      });
+
+      /*Tests_SRS_NODE_DEVICE_AMQP_16_038: [The `enableMethods` method shall connect and authenticate the transport if it is disconnected.]*/
+      it('calls the callback with an error if the transport fails to connect', function (testCallback) {
+        fakeBaseClient.connect = sinon.stub().callsArgWith(2, new Error('fake error'));
+        transport.enableMethods(function (err) {
+          assert(fakeBaseClient.connect.calledOnce);
+          assert.instanceOf(err, Error);
+          testCallback();
+        });
+      });
+
+      /*Tests_SRS_NODE_DEVICE_AMQP_16_038: [The `enableMethods` method shall connect and authenticate the transport if it is disconnected.]*/
+      it('calls the callback with an error if the transport fails to initialize the CBS links', function (testCallback) {
+        fakeBaseClient.initializeCBS = sinon.stub().callsArgWith(0, new Error('fake error'));
+        transport.enableMethods(function (err) {
+          assert(fakeBaseClient.connect.calledOnce);
+          assert(fakeBaseClient.initializeCBS.calledOnce);
+          assert.instanceOf(err, Error);
+          testCallback();
+        });
+      });
+
+      /*Tests_SRS_NODE_DEVICE_AMQP_16_038: [The `enableMethods` method shall connect and authenticate the transport if it is disconnected.]*/
+      it('calls the callback with an error if the transport fails to make a new putToken request on the CBS links', function (testCallback) {
+        fakeBaseClient.putToken = sinon.stub().callsArgWith(2, new Error('fake error'));
+        transport.enableMethods(function (err) {
+          assert(fakeBaseClient.connect.calledOnce);
+          assert(fakeBaseClient.initializeCBS.calledOnce);
+          assert(fakeBaseClient.putToken.calledOnce);
+          assert.instanceOf(err, Error);
+          testCallback();
+        });
+      });
+
+      /*Tests_SRS_NODE_DEVICE_AMQP_16_039: [The `enableMethods` method shall attach the method links and call its `callback` once these are successfully attached.]*/
+      it('attaches the method links', function (testCallback) {
+        transport.connect(function () {
+          assert(fakeBaseClient.attachReceiverLink.notCalled);
+          transport.enableMethods(function () {
+            assert(fakeBaseClient.attachSenderLink.calledWith(transport._deviceMethodClient._methodEndpoint));
+            assert(fakeBaseClient.attachReceiverLink.calledWith(transport._deviceMethodClient._methodEndpoint));
+            testCallback();
+          });
+        });
+      });
+
+      /*Tests_SRS_NODE_DEVICE_AMQP_16_040: [The `enableMethods` method shall call its `callback` with an `Error` if the transport fails to connect, authenticate or attach method links.]*/
+      // TODO: fix Twin code that emits error instead of calling callback.
```

Comment:
>// TODO: fix Twin code that emits error instead of calling callback [](start = 6, length = 67)

Different codeflow?

Reply:
yup

---
In reply to: [142282758](https://github.com/Azure/azure-iot-sdk-node/pull/97#discussion_r142282758) [](ancestors = 142282758)